Implement db helper method to retrieve available listings by market and date range

christine-w opened the pull request.

christine-w committed:
Update project directory structure to organize setup scripts
Also added csv files to ,gitignore

christine-w committed:
Update method names to be more concise and added npm script for db setup

christine-w committed:
Refactor inventory dbSetup script to import files for all markets
Please note that this is a work in progress (i.e., not yet fully functional) and committed for review purposes.

christine-w committed:
Finish refactor of inventory db setup script to import more than one file

christine-w committed:
Retrieve available listings from the db by market and date range

christine-w committed:
Retrieve available listings from the db by market and date range

christine-w committed:
Add unit tests for db helper to retrieve listings by market and date range

christine-w committed:
Add unit tests for db helper to retrieve listings by market and date range

christine-w committed:
Merge pull request #8 from model-airbnb/feat/10MInventory

Update inventory db setup script to import files for multiple markets

tylertruong commented on inventoryLocalStore/index.js:
```diff
@@ -1,10 +1,11 @@
 
+const getLengthOfStay = (start, end) =>
+  ((new Date(end) - new Date(start)) / 86400000) + 1;
+
```
Might be worth a code explanation for later. Hard to see that 86400000 = 1 day.

christine-w replied:
Noted. I will update PR.

tylertruong commented on inventoryLocalStore/index.js:
```diff
@@ -14,6 +15,29 @@ class Inventory {
+    const queryLimit = limit ? ` FETCH FIRST ${limit * getLengthOfStay(start, end)} ROWS ONLY` : '';
+    const queryString = `
+      SELECT DISTINCT *
+      FROM listings l
```
Clarify what l is or make it more descriptive.

christine-w replied:
OK, I'll add the "AS" syntax to make the aliasing more clear.

tylertruong reviewed (commented):
This is great! 👍 

I need to review database queries so did not test your query as extensively as I should. Perhaps @tiffaphm or @ryanakiyama can give a second opinion.

christine-w committed:
Merge branch 'feat/dateRangeDbHelper' of https://github.com/model-airbnb/Search into feat/dateRangeDbHelper

christine-w committed:
Address code review feedback
- Clarified ms in a day with a constant
- Added "AS" in SQL query when aliasing for clarity
- Removed Fakecity unit test for market and date range
